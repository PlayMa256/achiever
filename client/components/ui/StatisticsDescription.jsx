import React from 'react';
import PropTypes from 'prop-types';

import strings from '../../../shared/strings';
import './StatisticsDescription.styl';

const StatisticsDescription = ({ description, value }) => (
	<span className='StatisticsDescription'>
		{description} <strong>{value}</strong>
	</span>
);

export default StatisticsDescription;

StatisticsDescription.propTypes = {
	description: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired
}
