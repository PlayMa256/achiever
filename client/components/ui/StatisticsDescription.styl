@import "../../styles/_colours.styl"
@import "../../styles/_standards.styl"

.StatisticsDescription
	font-size: 11px
	color: transitionColour2;
	display: block;
	strong
		color: transitionColour1;
		margin-left: 4px;