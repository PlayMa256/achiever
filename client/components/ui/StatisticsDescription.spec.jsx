import React from 'react';
import { shallow } from 'enzyme';

import StatisticsDescription from './StatisticsDescription';

describe('StatisticsDescription', () => {
	it('should render with a properly', () => {
		const wrapper = shallow(<StatisticsDescription description='Some description' value='08:00' />);
		expect(wrapper).toMatchSnapshot();
	});
});
