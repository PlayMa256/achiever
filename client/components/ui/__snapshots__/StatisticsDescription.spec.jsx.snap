// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsDescription should render with a properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatisticsDescription
    description="Some description"
    value="08:00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Some description",
        " ",
        <strong>
          08:00
        </strong>,
      ],
      "className": "StatisticsDescription",
    },
    "ref": null,
    "rendered": Array [
      "Some description",
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "08:00",
        },
        "ref": null,
        "rendered": "08:00",
        "type": "strong",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Some description",
          " ",
          <strong>
            08:00
          </strong>,
        ],
        "className": "StatisticsDescription",
      },
      "ref": null,
      "rendered": Array [
        "Some description",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "08:00",
          },
          "ref": null,
          "rendered": "08:00",
          "type": "strong",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
